azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -210,38 +221,112 @@ private void createLogger() {
               + this.jobId;
       logger = Logger.getLogger(loggerName);
 
-      // Create file appender
-      String logName = createLogFileName(node);
-      logFile = new File(workingDir, logName);
-
-      String absolutePath = logFile.getAbsolutePath();
-
-      jobAppender = null;
+      // Attempt to create appenders as necessary
       try {
-        RollingFileAppender fileAppender =
-            new RollingFileAppender(loggerLayout, absolutePath, true);
-        fileAppender.setMaxBackupIndex(jobLogBackupIndex);
-        fileAppender.setMaxFileSize(jobLogChunkSize);
-        jobAppender = fileAppender;
-        logger.addAppender(jobAppender);
-        logger.setAdditivity(false);
+        attachFileAppender(createFileAppender());
       } catch (IOException e) {
+        removeAppender(jobAppender);
         flowLogger.error("Could not open log file in " + workingDir
             + " for job " + this.jobId, e);
       }
+
+      if (!azkabanProps.getBoolean(Constants.AZKABAN_SERVER_LOGGING_KAFKA_GLOBAL_DISABLE, false) &&
```

Comment:
This has an added benefit of dealing with the situation when the admins forgot to config one of the two server configs more explicitly and make it easier to debug.